pilot/pkg/model/xds_cache.go

```diff
@@ -164,17 +174,46 @@ func newLru(evictCallback simplelru.EvictCallback) simplelru.LRUCache {
 	return l
 }
 
-func (l *lruCache) evict(k interface{}, v interface{}) {
+// This is the callback passed to LRU, it will be called whenever a key is removed.
+func (l *lruCache) evict(k interface{}, _ interface{}) {
 	if features.EnableXDSCacheMetrics {
 		xdsCacheEvictions.Increment()
 	}
 
-	key := k.(string)
-	value := v.(cacheValue)
+	if l.recordEvicted {
+		l.evictedKeys.Insert(k.(string))
+	}
+}
 
-	// we don't need to acquire locks, since this function is called when we write to the store
-	l.clearConfigIndex(key, value.dependentConfigs)
-	l.clearTypesIndex(key, value.dependentTypes)
+// clearEvicted is run periodically to clear configIndex and typesIndex
+func (l *lruCache) clearEvicted() {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+	// In order to save cpu, if the evicted keys is very small, we can just ignore
+	// TODO: make it configurable
+	if len(l.evictedKeys) < 100 {
```

Comment:
That is worth discussion, but not critical since i checked it. I can use signal condition or channel to notify.

Reply:
May be move this up to handle func so that when it comes here we actually delete ? and would it make sense to have this value relative to cache size (some %age of cache size )?